universe/syncer.go

```diff
@@ -572,7 +592,41 @@ func (s *SimpleSyncer) SyncUniverse(ctx context.Context, host ServerAddr,
 
 	// With the engine created, we can now sync the local Universe with the
 	// remote instance.
-	return s.executeSync(ctx, diffEngine, syncType, syncConfigs, idsToSync)
+	diffs, err := s.executeSync(
+		ctx, diffEngine, syncType, syncConfigs, idsToSync,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute sync: %w", err)
+	}
+
+	// Notify subscribers of the new sync diffs where the old and
+	// new Universe roots differ.
+	var events []fn.Event
+	for idx := range diffs {
+		diff := diffs[idx]
+
+		// Only notify subscribers if the old and new Universe roots
+		// differ.
+		rootChange, err := diff.HasRootChanged()
+		if err != nil {
+			return nil, fmt.Errorf("unable to determine "+
+				"if root has changed: %w", err)
+		}
+
+		if rootChange {
+			events = append(
+				events,
+				&SyncDiffEvent{
+					timestamp: time.Now().UTC(),
+					SyncDiff:  diff,
+				},
+			)
+		}
+	}
+
+	s.eventDistributor.NotifySubscribers(events...)
```

Comment:
Assuming we're planning to use this to drive events we send to the supply commit state machines:
  * This only covers the new issuance events. 
  * Burn events won't show up here in the normal universe trees, nor will ignore events. 

We should also consider persistence and durability here. As an example: what happens if we emit this event, but then crash before it reaches a supply commit state machine? Some sort of log may be in order here. 

For burn+ignore, I think this new supply commit syncer will need to rely on polling for now against the specified universe sever.